src/io/gaf.rs

```diff
@@ -0,0 +1,617 @@
+            terminator: String::from_utf8(vec![termi]).unwrap(),
+            fileformat: fileformat,
+        }
+    }
+
+    /// Write a given GAF record.
+    pub fn write(&mut self, record: &Record) -> csv::Result<()> {
+        let db_ref = if !record.db_ref.is_empty() {
+            record
+                .db_ref
+                .iter()
+                .map(|(a, b)| format!("{}{}{}", a, self.delimiter, b))
+                .join(&self.terminator)
+        } else {
+            "".to_owned()
+        };
+
+        let qualifier = if !record.qualifier.is_empty() {
+            record.qualifier.join(&self.terminator.to_string())
+        } else {
+            "".to_owned()
+        };
+
+        let with_from = if !record.with_from.is_empty() {
+            record.with_from.join(&self.terminator.to_string())
+        } else {
+            "".to_owned()
+        };
+
+        let db_object_synonym = if !record.db_object_synonym.is_empty() {
+            record.db_object_synonym.join(&self.terminator.to_string())
+        } else {
+            "".to_owned()
+        };
+
+        let taxon = if !record.taxon.is_empty() {
+            record.taxon.join(&self.terminator.to_string())
+        } else {
+            "".to_owned()
+        };
+
+        let annotation_extension = if !record.annotation_extension.is_empty() {
+            record
+                .annotation_extension
+                .join(&self.terminator.to_string())
+        } else {
+            "".to_owned()
+        };
+
+        let mut v = vec![
+            record.db.clone(),
+            record.db_object_id.clone(),
+            record.db_object_symbol.clone(),
+            qualifier,
+            record.go_id.clone(),
+            db_ref,
+            record.evidence_code.clone(),
+            with_from,
+            record.aspect.clone(),
+            record.db_object_name.clone(),
+            db_object_synonym,
+            record.db_object_type.clone(),
+            taxon,
+            record.date.clone(),
+            record.assigned_by.clone(),
+        ];
+        if self.fileformat.name() == "GAF2" {
+            v.push(annotation_extension);
+            v.push(record.gene_product_form_id.clone());
+        }
+        self.inner.serialize(v)
+    }
+}
+
+/// A GAF record
+#[derive(Debug, Default, Serialize, Deserialize, Clone)]
+pub struct Record {
+    db: String,
+    db_object_id: String,
+    db_object_symbol: String,
+    qualifier: Vec<String>,
+    go_id: String,
+    db_ref: MultiMap<String, String>,
+    evidence_code: String,
+    with_from: Vec<String>,
+    aspect: String,
+    db_object_name: String,
+    db_object_synonym: Vec<String>,
+    db_object_type: String,
+    taxon: Vec<String>,
+    date: String,
+    assigned_by: String,
+    annotation_extension: Vec<String>,
+    gene_product_form_id: String,
+}
+
+impl Record {
+    /// Create a new GAF record.
+    pub fn new() -> Self {
+        Record {
+            db: "".to_owned(),
```

Comment:
Instead of `"".to_owned()`, I think the more generally used method here would be [`String::new()`](https://doc.rust-lang.org/std/string/struct.String.html#method.new). It also makes things a bit clearer to others reading the code.

Reply:
Done.